trade-finance-manager-ui/server/apis/entra-id.api.ts

```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+import { EntraIdConfig } from '../configs/entra-id.config';
+
+type GetAuthCodeUrlResponse = unknown;
+
+export class EntraIdApi {
+  private readonly entraIdConfig: EntraIdConfig;
+
+  constructor({ entraIdConfig }: { entraIdConfig: EntraIdConfig }) {
+    this.entraIdConfig = entraIdConfig;
+  }
+
+  public async getAuthorityMetadataUrl(): Promise<GetAuthCodeUrlResponse> {
+    const response = await axios.get<GetAuthCodeUrlResponse>(this.entraIdConfig.authorityMetadataUrl);
+    return response.data;
+  }
+}
```

Comment:
I'm wondering if this should be a different directory, the following could be confusing

```ts
/server/apis/
/server/api
```

Reply:
Thanks @AlexBramhill - what do you think about, maybe `third-party-apis` or `external-api`? @abhi-markan ?

I'm hesitant about `external-api` as we already have a microservice with the same name (great, makes sense)

I just feel that `apis` is too generic - concerned a future developer could go into here thinking "this is where the UI makes all the API calls", but it's not, 99.9% of them happen in `api.js`

Reply:
Happy to go for third-party-apis @abhi-markan ?

Reply:
I'm going to make this change and push to start on the next ticket, but I'll leave this thread open in case there's additional comment